### DISCUSSION TITLE
Very Custom Results Set

### DISCUSSION THREAD

**Discussion by cbdg-mitch** [Blazored/Typeahead#268] [Q&A]
Greetings @chrissainty,

I have a working Typeahead here (works great!), but I'd love to make my results set split up by Record Type. In my TypeAheadResult object, there can be "People", "Contacts" or "Companies", all combined into a generic data set with a Type string to differentiate. Each of these data sets come from API data.

I would like my results set when searching to be like:

People
--- A 
--- B
--- C

Contacts
--- A 
--- B
--- C

Companies
--- A 
--- B
--- C

```
@using BscModel.TypeAhead;

<BlazoredTypeahead SearchMethod="SearchRecords"
                   TValue="TypeAheadResult"
                   TItem="TypeAheadResult"
                   Value="selectedRecord"
                   ValueChanged="@( (TypeAheadResult s) => SelectedResultChanged(s) )"
                   ValueExpression="@( () => selectedRecord )"
                   Placeholder="Start searching ...">
    <SelectedTemplate>
        @context.RecordName
    </SelectedTemplate>
    <ResultTemplate>
        @context.RecordName (@context.RecordType)
    </ResultTemplate>
</BlazoredTypeahead>

@code {
    public TypeAheadResult selectedRecord;
    public List<Person> people;

    //public List<Company> companies;
    //public List<Contact> contacts;

    private IEnumerable<TypeAheadResult> typeAheadResults;

    protected override async void OnInitialized() {
        PersonSearch entity = new PersonSearch();
        entity.OrganizationId = loginState.OrganizationId;
        entity.IsActive = true;

        people = (await PersonService.GetAllPeople(entity)).ToList();

        typeAheadResults = people.Select(x => new TypeAheadResult { RecordId = x.PersonId, RecordType = "Person", RecordName = x.FullName });
    }

    private async Task SelectedResultChanged(TypeAheadResult result) {
        if (result == null) {
            return;
        }

        selectedRecord = result;

        if (selectedRecord.RecordType == "Person") {
            NavManager.NavigateTo("/PersonDetail/" + selectedRecord.RecordId);
        }
    }

    private async Task<IEnumerable<TypeAheadResult>> SearchRecords(string searchText) {
        return await Task.FromResult(typeAheadResults.Where(x => x.RecordName.ToLower().Contains(searchText.ToLower())));
    }
}```

**Comment by 0ptim** [comment]
One way would be to just have a condition in your `ResultTemplate` depending on an Enum or Type. Depending on this you can display different result UI.

```
<ResultTemplate>
    @if (MyType == People)
    {
            
    }
    else if (MyType == Contacts)
    {
            
    }
    else if (MyType == Companies)
    {
            
    }
</ResultTemplate>
```

**Reply by cbdg-mitch** [reply]
I'll give this a try. The only thing I thought about here was that the resulttemplate was a "per item" basis and it wouldn't actually group the records and instead just repeat the group header for each item. I'll try it out and report back.

**Reply by cbdg-mitch** [reply]
Yeah, as I thought, the ResultTemplate renders per item, so it doesn't allow for grouping the results by type. @chrissainty any thoughts on this? Might be a cool feature request to be able to group results by a property.

**Reply by 0ptim** [reply]
I did not understand your initial question right, sorry. Maybe another way is:

1. Get the data from wherever you're getting your data from.
2. Create a List<MyData> which has every possible result in it.
3. Order the List for your Types (All people, all contacts and then all companies)
4. Cycle through the list in Razor as I described above.
5. On every iteration, check if the type of your item has changed and if so, display an additional header as you wish.

I hope this is somewhat understandable. Give it a go and report back please.
